### DISCUSSION TITLE
Flutter | model_viewer_plus | Online objects from Web URL

### DISCUSSION THREAD

**Discussion by ladonskliarov** [google/model-viewer#4460] [Q&A]
Where I can place my 3D-model(glb) in online to read it in the next part of code?

I tried to place it here:
**GitHub**
https://github.com/ladonskliarov/test/blob/04c0b18e0b732f3f7d4f95891eee8d0ec9092a2c/Astronaut.glb
**Firebase Storage**
https://firebasestorage.googleapis.com/v0/b/personal-gym-ukr.appspot.com/o/poly.glb?alt=media&token=b0b10db8-e97b-4a8c-ac36-eafcddf8d928

**Code part here:**
body: const ModelViewer(
          backgroundColor: Color.fromARGB(0xFF, 0xEE, 0xEE, 0xEE),
          src: MY OBJECT SHOULD BE HERE,
          alt: 'A 3D model of an astronaut',
          ar: true,
          arModes: ['scene-viewer', 'webxr', 'quick-look'],
          autoRotate: true,
          iosSrc: 'https://modelviewer.dev/shared-assets/models/Astronaut.usdz',
          disableZoom: true,
        ),
        
I see nothing. There isn't any uploading under AppBar.        
Thank you in advance!

**Comment by elalish** [comment]
Have you checked the JS console for errors? The most common issue is lacking CORS headers, which you set server side.

**Reply by Kunal0127** [reply]
How can we do that can someone explain please?
coz i guess i have same issue


D/Surface (29171): lockHardwareCanvas
I/chatty  (29171): uid=10261(com.example.scanner_app) identical 5 lines
D/Surface (29171): lockHardwareCanvas
I/chromium(29171): [INFO:CONSOLE(0)] "Access to fetch at 'https://firebasestorage.googleapis.com/v0/b/eduar-976af.appspot.com/o/models%2Ftabletop_football.glb?alt=media&token=623e02ffc3e-4a03-84df-94f893ba13c9' (redirected from 'http://127.0.0.1:4061/model') from origin 'http://127.0.0.1:4061' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.", source: http://127.0.0.1:4061/ (0)
I/chromium(29171): [INFO:CONSOLE(557)] "TypeError: Failed to fetch", source: http://127.0.0.1:40561/model-viewer.min.js (557)
I/ple.scanner_ap(29171): Background concurrent copying GC freed 41671(2MB) AllocSpace objects, 0(0B) LOS objects, 50% free, 2MB/5MB, paused 377us total 120.761ms
I/ple.scanner_ap(29171): Background concurren

**Reply by elalish** [reply]
https://modelviewer.dev/docs/faq.html#entrydocs-technical-questions-cors

**Comment by Vegi1** [comment]
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.black,
        accentColor: Colors.anthracite,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final _passwordController = TextEditingController();

  void _login() {
    if (_passwordController.text.length >= 8) {
      //TODO: implement login logic here
      print('Login successful');
      Navigator.pushReplacement(
        context,
        MaterialPageRoute(builder: (context) => MyMainPage()),
      );
    } else {
      print('Password must be at least 8 characters long');
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Login'),
      ),
      body: Padding(
        padding: const EdgeInsets.all(20),
        child: Column(
          children: [
            TextField(
              controller: _passwordController,
              obscureText: true,
              decoration: InputDecoration(
                labelText: 'Password',
                border: OutlineInputBorder(),
              ),
            ),
            SizedBox(height: 20),
            ElevatedButton(
              onPressed: _login,
              child: Text('Log in'),
            ),
          ],
        ),
      ),
    );
  }
}

class MyMainPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Main Page'),
      ),
      body: Center(
        child: Text('Welcome to the main page!'),
      ),
    );
  }
}
